examples/megacli.cpp

```diff
@@ -6684,12 +6689,59 @@ void DemoApp::userattr_update(User* u, int priv, const char* n)
           << n << " added or updated" << endl;
 }
 
+#ifndef NO_READLINE
+char* longestCommonPrefix(ac::CompletionState& acs)
+{
+    string s = acs.completions[0].s;
+    for (int i = acs.completions.size(); i--; )
+    {
+        for (unsigned j = 0; j < acs.completions[i].s.size(); ++j)
+        {
+            if (j < s.size() && s[j] != acs.completions[i].s[j])
```

Comment:
Minor improvement (untested):
```
        for (unsigned j = 0; j < acs.completions[i].s.size() && j < s.size(); ++j)
        {
            if (s[j] != acs.completions[i].s[j])
```

Reply:
Excellent point, I've made that improvement, and tested it on linux and windows.   I noticed one other thing while testing as well, when trying to iterate a non-existent directory, and that is fixed also.  thanks!